### PR TITLE
Check that an account is OAuth linked before querying permissions

### PR DESCRIPTION
**Pull request by FastLizard4**
Alternate fix for enwikipedia-acc/waca#221

### CODE REVIEW

File: includes/DataObjects/User.php [hunk 1]
```diff
@@ -822,8 +825,15 @@ public function oauthCanCreateAccount()
 		}
 	}
 
+	/**
+	 * @return bool
+	 */
 	public function oauthCanCheckUser()
```
Comment (stwalkerster) [inline includes/DataObjects/User.php]: may be worth making this protected if it's only every called from here and CommunityUser.

Comment (FastLizard4) [inline includes/DataObjects/User.php]: Hrm, that might be quite a rabit hole though.  There's probably a *lot* of stuff we can make more private in various places.

Comment (stwalkerster) [inline includes/DataObjects/User.php]: True. Again this is OAuth code which I dislike. Meh.

:shipit:

Comment (FastLizard4) [inline includes/DataObjects/User.php]: On the other hand, we might as well start somewhere.  Made oauthCanCheckUser protected.

**Comment by FastLizard4** [comment]
YOLO

**Pull request merged**